### PR TITLE
Code Quality: Introduced ComHeapPtr

### PR DESCRIPTION
**Pull request by 0x5bfa**
### Resolved / Related Issues

- Related #16071
- Related #15000

### Steps used to test these changes

_None_

### CODE REVIEW

**Comment by 0x5bfa** [comment]
As far as we enable friendlyOverloads and useSafeHandles we don't have to worry about handles and we should use SafeHandle derivatives.

@hez2010 fyi.

**Comment by 0x5bfa** [comment]
@hez2010 Let me know what you think of ComQIPtr which automatically QI's when initialized with a existing com object and disposes when no longer needed, when you got some time to review this:)

```c#
ComPtr<IShellItem> pShellItem = default;
...
ComQIPtr<IShellItem2> pShellItem2 = new(pShellItem);
```

File: src/Files.App.CsWin32/Windows.Win32.ComHeapPtr.cs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+{
+	/// <summary>
+	/// Contains a heap pointer allocated via CoTaskMemAlloc and a set of methods to work with the pointer safely.
+	/// </summary>
+	public unsafe struct ComHeapPtr<T> : IDisposable where T : unmanaged
+	{
+		private T* _ptr;
+
+		public bool IsNull
+			=> _ptr == default;
```
Comment (hez2010) [inline src/Files.App.CsWin32/Windows.Win32.ComHeapPtr.cs:19]: You can use `null` here.

**Review by hez2010 — APPROVED** [review]
Looks great.

File: src/Files.App.CsWin32/Windows.Win32.ComHeapPtr.cs (hunk 1 above)
Comment (0x5bfa) [reply src/Files.App.CsWin32/Windows.Win32.ComHeapPtr.cs:19]: Which is good?

Comment (0x5bfa) [reply src/Files.App.CsWin32/Windows.Win32.ComHeapPtr.cs:19]: Which is better?

**Pull request merged**